Amazon.QLDB.Driver.IntegrationTests/AsyncStatementExecutionTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.QLDB.Driver.AsyncIntegrationTests
+{
+    using Amazon.QLDB.Driver.AsyncIntegrationTests.utils;
+    using Amazon.QLDB.Driver.IntegrationTests.utils;
+    using Amazon.QLDBSession;
+    using Amazon.QLDBSession.Model;
+    using Amazon.IonDotnet.Builders;
+    using Amazon.IonDotnet.Tree;
+    using Amazon.IonDotnet.Tree.Impl;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    [TestClass]
+    public class StatementExecutionTests
+    {
+        private static readonly IValueFactory ValueFactory = new ValueFactory();
+        private static AmazonQLDBSessionConfig amazonQldbSessionConfig;
+        private static AsyncIntegrationTestBase integrationTestBase;
+        private static AsyncQldbDriver qldbDriver;
+
+        [ClassInitialize]
+        public async static void SetUp(TestContext context)
+        {
+            // Get AWS configuration properties from .runsettings file.
+            string region = context.Properties["region"].ToString();
+
+            amazonQldbSessionConfig = AsyncIntegrationTestBase.CreateAmazonQLDBSessionConfig(region);
+            integrationTestBase = new AsyncIntegrationTestBase(Constants.LedgerName, region);
+
+            integrationTestBase.RunForceDeleteLedger();
+
+            integrationTestBase.RunCreateLedger();
+            qldbDriver = integrationTestBase.CreateDriver(amazonQldbSessionConfig);
+
+            // Create table.
+            var query = $"CREATE TABLE {Constants.TableName}";
+
+            var count = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(query);
+
+                var count = 0;
+                await foreach (var row in result)
+                {
+                    count++;
+                }
+                return count;
+            });
+
+
+            Assert.AreEqual(1, count);
+
+            var result = await qldbDriver.ListTableNamesAsync();
+            foreach (var row in result)
```

Comment:
await foreach

Reply:
The 'result' object here is an IEnumerable<string> which cannot be awaited.  This is because in AsyncQldbDriver the method is defined:
public async Task<IEnumerable<string>> ListTableNamesAsync(CancellationToken cancellationToken = default)

Reply:
With Simon's updated AsyncQldbDriver stuff, the result should become IAsyncEnumerable. Likewise for the other similar comments.